### ISSUE TITLE
"Failed to load data..." when selecting cluster

### ISSUE THREAD

**Issue by CapKenR** [walmartlabs/kubeman#3]
I'm running the 0.5 release on Windows 10 Pro Insider Edition. When I try select a cluster it gives me "Failed to load data for the following cluster(s): ...". I tried it with my local Docker Desktop "cluster" as well as two remote clusters each with two users. I can use kubectl to interact normally with all of them. By the end, the selection window said (in red):

Failed to load data for the following cluster(s): docker-desktop, ucp_test-aws-ucp.lab.capstonec.net:6443_admin, ucp_test-azure-ucp.lab.capstonec.net:6443_admin, ucp_test-aws-ucp.lab.capstonec.net:6443_kenrider, ucp_test-azure-ucp.lab.capstonec.net:6443_kenrider

**Comment by CapKenR** [comment]
I tried kubeman on another laptop with only one cluster and user configured. I get the same error. I don't see any log files specific to kubeman and I don't see anything related in the Windows event logs.

**Comment by uk000** [comment]
Error handling and error reporting is not implemented well in kubeman so far, could be done better. For now your best bet to investigate this would be to open the debug console. You can do that by selecting the menu option under debug in the menu bar.  With the console open, if you reselect the cluster, it should show some exception in the console. You could check if it logs something related to permissions (e.g. 403 forbidden). That could give you a clue about whether this is related to permissions given to API based k8s access, which has its own cluster permission settings different than what's needed for kubectl access.

**Comment by CapKenR** [comment]
Of course, now that I see the (obvious) way to debug, I can't seem to duplicate the problem. I do have another problem but it's unrelated so I'll open a separate issue.

**Comment by CapKenR** [comment]
Reopening as I ran into this problem again and have the debug console output for it. The cluster/context had been working just a few minutes ago but now I'm getting the "Failed to load data..." error again.

Here's the debug console output for `renderer.js:1201`.

```
Error while loading namespaces for cluster ucp_test-ucp.lab.capstonec.net:6443_admin: Error: getaddrinfo ENOTFOUND test-ucp.lab.capstonec.net test-ucp.lab.capstonec.net:6443
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:57:26)
```

**Comment by CapKenR** [comment]
BTW, `kubectl` commands with this context are working fine.

**Comment by uk000** [comment]
The error suggests that DNS resolution may have failed for the fqdn. Can you do a nslookup/dig from your local machine to check if the fqdn `ucp_test-ucp.lab.capstonec.net` resolves?

**Comment by CapKenR** [comment]
`ucp_test-ucp.lab.capstonec.net` is part of the context name and not resolvable. The server name in the cluster spec is https://test-ucp.lab.capstonec.net:6443. I can resolve test-ucp.lab.capstonec.net (it's a CNAME for a load balancer fronting the cluster) and can use `kubectl` with the cluster at the time Kubeman has the problem.

**Comment by uk000** [comment]
Yeah, the error says 'getaddrinfo ENOTFOUND test-ucp.lab.capstonec.net', so either the kubernetes API client wasn't able to resolve the DNS or may be unable to connect due to some other reasons. But you said in a previous comment that it also worked at some point, right?

**Comment by CapKenR** [comment]
It worked the first time then sometime later in a separate start of the kubeman executable it didn’t. It’s done that several times. Every time it didn’t work in kubeman, it did work using kubectl from the command line.
